templates/tftmpl/source_input.go

```diff
@@ -0,0 +1,96 @@
+package tftmpl
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/hashicorp/consul-terraform-sync/logging"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+)
+
+var (
+	_ SourceInput = (*ServicesSourceInput)(nil)
+)
+
+// SourceInput handles appending a run source input's relevant templating for Terraform
+// generated files
+type SourceInput interface {
+	// ServicesAppended returns true if the services variable has been appended
+	// to the template content.
+	ServicesAppended() bool
+
+	// appendModuleAttribute writes to an HCL module body the SourceInput variable
+	// as a module argument in main.tf file.
+	// module "name" {
+	//   services = var.services
+	// }
+	appendModuleAttribute(*hclwrite.Body)
+
+	// appendTemplate writes the generated variable template for the SourceInput
+	// based on whether the source includes the source variable.
+	appendTemplate(io.Writer) error
+
+	// appendVariable writes the corresponding Terraform variable block to
+	// the variables.tf file.
+	appendVariable(io.Writer) error
+}
+
+// ServicesSourceInput handles appending a run source variable's relevant templating for Terraform
+// generated files
+type ServicesSourceInput struct {
```

Comment:
I have a similar concern for the `ServicesSourceInput` object in relation to [`ServicesCondition`](https://github.com/hashicorp/consul-terraform-sync/blob/master/templates/tftmpl/condition.go#L52) as the config comment for `SourceInputConfig`.

We have 2-3 more known source inputs after services regex. To support them in the same manner as this would have significant code overlap. Could you look into refactoring it in such a way where source inputs and conditions could use the same base?